.github/workflows/ci.yaml

```diff
@@ -14,30 +14,101 @@ env:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+          - {
```

Comment:
Consider this syntax rather than the braces.

```yaml
config:
  - os: "ubuntu-latest",
    extension: "",
    platformAgnosticChecks: true
  - os: "macos-latest",
    extension: "",
    extraArgs: ""
  - os: "windows-latest",
    extension: ".exe",
    extraArgs: ""
```